src/matlab2tikz.m

```diff
@@ -3930,7 +3930,11 @@ function warnFacetedInterp(m2t)
             'scatter/@pre marker code/.append style', ...
             '{/tikz/mark size=\perpointmarksize}');
     end
-
+    % Add forget plot
+    if ~m2t.currentHandleHasLegend 
+        hiddenDrawOptions = maybeShowInLegend(false, drawOptions);
+        drawOpts = opts_print(hiddenDrawOptions);
```

Comment:
I don't think this can work: `drawOpts` is overwritten later on (line 3940/3944), so I am pretty sure your change does not affect the produced output.

What, I think, could work instead of your change:
```matlab
drawOptions = maybeShowInLegend(m2t.currentHandleHasLegend, drawOptions);
```
at the exact location where you put the code.
There is no need for `opts_print` (that is meant for outputting as string), and no need for an `if`, since the function you are calling is:
```matlab
function [drawOptions] = maybeShowInLegend(showInLegend, drawOptions)
    % sets the appropriate options to show/hide the plot in the legend
    if ~showInLegend
        % No legend entry found. Don't include plot in legend.
        drawOptions = opts_add(drawOptions, 'forget plot');
    end
end
```

Thank you for addressing this oversight! 👍  Could you try out with the suggested fix, to see whether that solves the problem you were facing?